tracing-subscriber/src/filter/directive.rs

```diff
@@ -0,0 +1,364 @@
+use crate::filter::level::{self, LevelFilter};
+use std::{cmp::Ordering, error::Error, fmt, iter::FromIterator, str::FromStr};
+use tracing_core::Metadata;
+/// Indicates that a string could not be parsed as a filtering directive.
+#[derive(Debug)]
+pub struct ParseError {
+    kind: ParseErrorKind,
+}
+/// A directive which will statically enable or disable a given callsite.
+///
+/// Unlike a dynamic directive, this can be cached by the callsite.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub(crate) struct StaticDirective {
+    pub(in crate::filter) target: Option<String>,
```

Comment:
In general, I try to avoid using `super` except in modules that are nested in the same file (like `tests` modules), because it's nice to be able to move the file around in the module tree and not have to fix imports and visibility modifiers.